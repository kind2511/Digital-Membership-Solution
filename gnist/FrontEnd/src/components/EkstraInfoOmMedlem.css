.eiom-container {
  background-color: #f5f5f5;
  border: 1px solid #ccc;
  border-radius: 8px;
  padding: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.eiom-title {
  color: #333;
  font-size: 20px;
  margin-bottom: 10px;
}

.eiom-search-input {
  width: 100%;
  padding: 10px;
  margin-bottom: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.eiom-search-status {
  color: #666;
  margin-bottom: 10px;
}

.eiom-section-content {
  background-color: #fff;
  padding: 10px;
  border-radius: 4px;
  max-height: 200px;
  overflow-y: auto;
}

.eiom-search-result {
  padding: 10px;
  border-bottom: 1px solid #eee;
  cursor: pointer;
}

.eiom-search-result:hover {
  background-color: #f7f7f7;
}

.eiom-search-result:last-child {
  border-bottom: none;
}

.eiom-modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
}

.eiom-modal-content {
  background-color: #fff;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  width: 80%;
  max-width: 500px;
  text-align: center;
}

.eiom-modal-content h3 {
  margin-bottom: 15px;
}

.eiom-modal-content p {
  margin-bottom: 20px;
}

/* Button styles */
.eiom-modal-buttons {
  display: flex;
  justify-content: center;
  gap: 20px; 
}

.eiom-modal-buttons .eiom-delete-btn {
  padding: 10px 20px;
  border: none;
  background-color: #d9534f; 
  color: white;
  border-radius: 5px;
  cursor: pointer;
}

.eiom-modal-buttons .eiom-delete-btn:hover {
  background-color: darkred;
}

.eiom-modal-buttons .eiom-close-btn {
  padding: 10px 20px;
  border: none;
  background-color: #005A9E; 
  color: white;
  border-radius: 5px;
  cursor: pointer;
}

.eiom-modal-buttons .eiom-close-btn:hover {
  background-color: #004a7c;
}
