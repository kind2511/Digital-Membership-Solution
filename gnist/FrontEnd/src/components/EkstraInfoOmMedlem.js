import React, { useState, useEffect } from 'react';
import './EkstraInfoOmMedlem.css';

function EkstraInfoOmMedlem() {
    const [searchTerm, setSearchTerm] = useState('');
    const [results, setResults] = useState([]);
    const [searchStatus, setSearchStatus] = useState('');
    const [selectedMember, setSelectedMember] = useState(null);
    const [showDeleteSuccessMessage, setShowDeleteSuccessMessage] = useState(false);

    useEffect(() => {
        const savedAuth0ID = localStorage.getItem('selectedAuth0ID');
        if (savedAuth0ID) {
            const member = results.find(m => m.auth0ID === savedAuth0ID);
            if (member) {
                setSelectedMember(member);
            }
        }
    }, [results]);

    const fetchData = (value) => {
        if (value.trim() === '') {
            setResults([]);
            setSearchStatus('');
            setSelectedMember(null);
            localStorage.removeItem('selectedAuth0ID');
            return;
        }
        setSearchStatus('Searching...');
        fetch(`http://127.0.0.1:8000/digital_medlemsordning/search_member/?name=${value}`)
            .then(response => {
                if (!response.ok) {
                    throw new Error(`HTTP error! status: ${response.status}`);
                }
                return response.json();
            })
            .then(data => {
                setResults(data);
                setSearchStatus(`Found ${data.length} results.`);
            })
            .catch(error => {
                console.error("Error fetching data:", error);
                setSearchStatus("Failed to fetch data. Please try again later.");
            });
    };

    const handleSearchChange = (e) => {
        const value = e.target.value;
        setSearchTerm(value);
        fetchData(value);
    };

    const handleSelectMember = (member) => {
        setSelectedMember(member);
        localStorage.setItem('selectedAuth0ID', member.auth0ID);
    };

    const handleDeleteMemberInfo = () => {
        const auth0ID = localStorage.getItem('selectedAuth0ID');
        if (auth0ID) {
            const url = `http://127.0.0.1:8000/digital_medlemsordning/remove_member_info/${auth0ID}/`;
            fetch(url, {
                method: 'PUT',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({})
            })
            .then(response => {
                if (!response.ok) {
                    throw new Error(`HTTP error! status: ${response.status}`);
                }
                return response.json();
            })
            .then(() => {
                setSelectedMember(null);
                localStorage.removeItem('selectedAuth0ID');
                fetchData(searchTerm);
                setShowDeleteSuccessMessage(true);
                setTimeout(() => setShowDeleteSuccessMessage(false), 3000);
            })
            .catch(error => {
                console.error("Error deleting member info:", error);
            });
        } else {
            console.error('Cannot delete member info: No auth0ID found in local storage');
        }
    };

    return (
        <div className="eiom-container">
            <h2 className="eiom-title">Ekstra Info om Medlem</h2>
            <input
                type="text"
                className="eiom-search-input"
                placeholder="SÃ¸k etter medlem..."
                value={searchTerm}
                onChange={handleSearchChange}
            />
            <div className="eiom-search-status">{searchStatus}</div>
            <div className="eiom-section-content">
                {results.map((member, index) => (
                    <div key={index} className="eiom-search-result" onClick={() => handleSelectMember(member)}>
                        {`${member.first_name} ${member.last_name}`}
                    </div>
                ))}
            </div>
            {selectedMember && (
                <div className="eiom-modal">
                    <div className="eiom-modal-content">
                        <h3>{`${selectedMember.first_name} ${selectedMember.last_name}`}</h3>
                        <p>{selectedMember.info || "N/A"}</p>
                        <div className="eiom-modal-buttons">
                            <button onClick={handleDeleteMemberInfo} className="eiom-delete-btn">Slett</button>
                            <button onClick={() => {
                                setSelectedMember(null);
                                localStorage.removeItem('selectedAuth0ID');
                            }} className="eiom-close-btn">Lukk</button>
                        </div>
                    </div>
                </div>
            )}
            {showDeleteSuccessMessage && (
                <div className="eiom-delete-success-banner">
                    Medlems ekstra info er slettet
                </div>
            )}
        </div>
    );
}

export default EkstraInfoOmMedlem;
