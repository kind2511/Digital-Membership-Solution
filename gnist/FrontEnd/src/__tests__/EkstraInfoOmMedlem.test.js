import React from 'react';
import renderer from 'react-test-renderer';
import EkstraInfoOmMedlem from '../components/EkstraInfoOmMedlem';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);
const membersMock = [
  {
    auth0ID: 'auth0|123456789',
    first_name: 'John',
    last_name: 'Doe',
    info: 'Some additional info about John Doe'
  },
  {
    auth0ID: 'auth0|987654321',
    first_name: 'Jane',
    last_name: 'Doe',
    info: 'Some additional info about Jane Doe'
  }
];
mock.onGet('http://127.0.0.1:8000/digital_medlemsordning/get_members_with_info/').reply(200, membersMock);

test('EkstraInfoOmMedlem component snapshot', () => {
  const component = renderer.create(<EkstraInfoOmMedlem />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
