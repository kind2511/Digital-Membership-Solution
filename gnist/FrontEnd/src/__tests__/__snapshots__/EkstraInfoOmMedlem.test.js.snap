// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EkstraInfoOmMedlem component snapshot 1`] = `
<div
  className="eiom-container"
>
  <h2
    className="eiom-title"
  >
    Ekstra Info om Medlem
  </h2>
  <div
    className="eiom-section-content"
  />
</div>
`;
